screwdriver/api/api.go

```diff
@@ -33,102 +38,220 @@ type ResponseError struct {
+	}
+
+	return nil
+}
+
+func writeMultipartBin(writer *multipart.Writer, commandSpec *util.CommandSpec) error {
+	// Load binary from file
+	fileContentsBin, err := util.LoadByte(commandSpec.Binary.File)
+	if err != nil {
+		return fmt.Errorf("Failed to load binary file:%v", err)
+	}
+
+	// Set filename of binary
+	fileNameBin := filepath.Base(commandSpec.Binary.File)
+	binPart, err := writer.CreateFormFile("file", fileNameBin)
+	if err != nil {
+		return fmt.Errorf("Failed to create form of binary:%v", err)
+	}
+
+	// Write binary part
+	_, err = binPart.Write(fileContentsBin)
+	if err != nil {
+		return fmt.Errorf("Failed to write binary to form:%v", err)
+	}
+
+	return nil
+}
+
+func (c client) httpRequest(httpMethod string, uri string, contentType string, body *bytes.Buffer) (*util.CommandSpec, error) {
+	// Send request
+	responseBytes, statusCode, err := c.sendHTTPRequest(httpMethod, uri, contentType, body)
+	if err != nil {
+		return nil, fmt.Errorf("Post request failed: %v", err)
+	}
+
+	// Check response
+	responseBytes, err = handleResponse(responseBytes, statusCode)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get response
+	responseSpec := &util.CommandSpec{}
+	err = json.Unmarshal(responseBytes, responseSpec)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse json response %v", err)
+	}
+
+	return responseSpec, err
+}
+
+func (c client) PostCommand(specPath string, commandSpec *util.CommandSpec) (*util.CommandSpec, error) {
+	// Generate payload
+	// Prepare body
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+
+	err := writeMultipartYaml(writer, specPath, commandSpec)
+	if err != nil {
+		return nil, err
+	}
+
+	switch commandSpec.Format {
+	case "binary":
+		err = writeMultipartBin(writer, commandSpec)
+	case "habitat":
+		return nil, fmt.Errorf("Posting habitat is not implemented yet")
+	case "docker":
+		return nil, fmt.Errorf("Posting docker is not implemented yet")
 	}
 
-	req.Header.Set("Content-type", "application/json")
-	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", c.jwt))
-	res, err := c.client.Do(req)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to get command from Screwdriver API: %v", err)
+		return nil, err
 	}
-	defer res.Body.Close()
-	body, err := handleResponse(res)
 
+	writer.Close()
+
+	// Generate URL
+	uri, err := url.Parse(c.baseURL)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse URL on POST: %v", err)
+	}
+	uri.Path = path.Join(uri.Path, "commands")
+
+	// Send request
+	contentType := writer.FormDataContentType()
+	responseSpec, err := c.httpRequest("POST", uri.String(), contentType, body)
 	if err != nil {
 		return nil, err
 	}
 
-	err = json.Unmarshal(body, cmd)
+	return responseSpec, nil
+}
+
+func (c client) sendHTTPRequest(method, url, contentType string, payload *bytes.Buffer) ([]byte, int, error) {
+	req, err := http.NewRequest(
+		method,
+		url,
+		payload,
+	)
```

Comment:
Why this code are consist of some lines?
I think you do not need to set new line
```
req, err := http.NewRequest(method,url,payload)
```

Reply:
Oh, it's my oversight. I've fixed. Thank you!